### PR TITLE
Add getters for controller objects

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add getters for controller objects to get their content easier on action hooks. Also, you don't have to re-intiailize them again.
| Type?             | refacto
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | -
| Related PRs       | -
| How to test?      | Hook to `displayHeader` with quality assurance module and try `dump($this->context->controller->getManufacturer());` on manufacturer detail page for example.
| Possible impacts? | No
| Sponsor company   | [TRENDO s.r.o.](https://www.trendo.cz)

### CODE REVIEW

**Review by nicosomb — COMMENTED** [review]
You can add phpdoc to give more details on return type for each function.

File: controllers/front/CmsController.php
Comment (zuk3975) [inline controllers/front/CmsController.php:242]: is there a point when property is public?

Comment (Hlavtox) [reply controllers/front/CmsController.php:242]: No, but I wanted to unify it, so we have one way of accessing this data by getters. And on the other side, I didn't want to break modules by setting the `$cms` as `protected`.

**Review by 0x346e3730 — DISMISSED** [review]

**Comment by zuk3975** [comment]
Im still not convinced that we need those methods for public properties, but if other @PrestaShop/committers are ok with that, then Im ok too.
Besides that all of your getters should have return types

**Review by matthieu-rolland — COMMENTED** [review]
If we are adding getters, but keeping the public properties for avoiding a BC break...

Then maybe we should deprecate the public properties so they can be set private in the next major version, and only getters would be allowed in the future, what do you think @zuk3975 and @Hlavtox ? 

(does it make sense, to deprecate a public property before changing its visibility to private ?)

**Comment by nicosomb** [comment]
@matthieu-rolland it's a good idea. But I don't know if it makes sense.

**Comment by Hlavtox** [comment]
Or I can just make everything public and getters also. 🤷‍♂️ A) We don't need it to be protected, B) Life is easier when you try something and it just works 😁

**Comment by FabienPapet** [comment]
> Or I can just make everything public ... Life is easier when you try something and it just works 😁

Depends for who, public properties can be a pain if you want to rename / deprecate / update some code, as there is no simple way to deprecate unless you call __get etc… 

> A) We don't need it to be protected

I think you are wrong on this, we need them to be read only. Therefore not public, but protected. Public properties are 💩 unless on a DTO, and controllers are not.

I'm 👍 on adding these methods, but we need to deprecate all the public, and replace all the current usages at the same time.

**Review by kpodemski — CHANGES_REQUESTED** [review]
I agree with @FabienPapet idea. It seems you should deprecate:
`$cms`, `$cms_category`.

You could also add return types to your functions :)

**Comment by Hlavtox** [comment]
Rebased, added docs, deprecated the public properties.
This doesn't need QA, it only adds getters.

File: controllers/front/CmsController.php
Comment (zuk3975) [inline controllers/front/CmsController.php]: could you mention the related getters which should be used now in every deprecation?
like 
```suggestion
     * @deprecated Since 8.1, it will become protected in next major version. Use getCms() method instead.
```
or maybe with a `@see tag`
```
 * @deprecated Since 8.1, it will become protected in next major version.
 * 
 * @see getCms()
```

both cases ok for me, as long as the alternative is mentioned

Comment (Hlavtox) [reply controllers/front/CmsController.php]: @zuk3975 Fixed ✅

**Review by zuk3975 — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
Waiting for CI + auto tests enabled:
https://github.com/kpodemski/ga.tests.ui.pr/actions/runs/3959621098

**Pull request merged**
